inc/Engine/Preload/Database/Queries/Cache.php

```diff
@@ -500,4 +509,22 @@ public function remove_all() {
 
 		$db->query( "DELETE FROM `$prefixed_table_name` WHERE 1 = 1" );
 	}
+
+	/**
+	 * Check if the url is rejected.
+	 *
+	 * @param string $url url to check.
+	 * @return bool
+	 */
+	protected function is_rejected( string $url ) {
+		$extensions = [
+			'php' => 1,
+			'xml' => 1,
+			'xsl' => 1,
```

Comment:
I'd suggest to add kml extension too, as there was a comment in the issue about this type